File: src/main/java/com/api/prisma_vi/color/ColorForm.java
package com.api.prisma_vi.color;

public record ColorForm(
        String name,
        String description,
        String[] psychologyTags,
        String designUsageSuggestions,
        SimpleColor[] colorPalette
) {
    public ColorForm() {
        this(
                "Descriptive_name_of_the_color,_e.g._'Navy_Blue'",
                "Brief_emotional_and_cultural_description_of_the_color._e.g._'Evokes_seriousness_and_confidence.'",
                new String[] { "e.g._trust", "e.g._seriousness", "e.g._stability" },
                "Suggestions_for_use_in_design_e.g._'Useful_as_a_background_color_in_corporate_apps'",
                new SimpleColor[] {
                        new SimpleColor("Name_of_the_complementary_or_analogous_color._e.g._'Light_Blue'", "e.g._#ADD8E6"),
                        new SimpleColor("Name_another_of_the_complementary_or_analogous_color._e.g._'Azul_Claro'", "e.g._#ADD8E6")
                }
        );
    }

    @Override
    public String toString() {

        return """
        {
          "name": "%s",
          "description": "%s",
          "psychologyTags": ["%s","%s","%s"],
          "designUsageSuggestions": "%s",
          "colorPalette": [
            {"name": "%s", "hex": "#%s"},
            {"name": "%s", "hex": "#%s"}
          ]
        }
        """.formatted(name,
                description,
                psychologyTags[0],
                psychologyTags[1],
                psychologyTags[2],
                designUsageSuggestions,
                colorPalette[0].name(),
                colorPalette[0].hex(),
                colorPalette[1].name(),
                colorPalette[1].hex());
    }
}

File: src/main/java/com/api/prisma_vi/color/ColorForm.kt
package com.api.prisma_vi.color

data class ColorForm(val name: String?,
                     val description: String?,
                     val psychologyTags: Array<String>,
                     val designUsageSuggestions: String?,
                     val colorPallete: Array<SimpleColor>) {
    constructor() : this(
        name = "Descriptive name of the color, e.g. 'Navy Blue'",
        description = "Brief emotional and cultural description of the color. e.g. 'Evokes seriousness and confidence.'",
        psychologyTags = arrayOf("e.g. trust", "e.g. seriousness", "e.g. stability"),
        designUsageSuggestions = "Suggestions for use in design e.g. 'Useful as a background color in corporate apps'",
        colorPallete = arrayOf(
            SimpleColor(
                name = "Name of the complementary or analogous color. e.g. 'Light Blue'",
                hex = "e.g. #ADD8E6"
            ),
            SimpleColor(
                name = "Name another of the complementary or analogous color. e.g. 'Azul Claro'",
                hex = "e.g. #ADD8E6"
            )
        )
    )
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ColorForm

        if (name != other.name) return false
        if (description != other.description) return false
        if (!psychologyTags.contentEquals(other.psychologyTags)) return false
        if (designUsageSuggestions != other.designUsageSuggestions) return false
        if (!colorPallete.contentEquals(other.colorPallete)) return false

        return true
    }

    override fun hashCode(): Int {
        var result = name?.hashCode() ?: 0
        result = 31 * result + (description?.hashCode() ?: 0)
        result = 31 * result + psychologyTags.contentHashCode()
        result = 31 * result + (designUsageSuggestions?.hashCode() ?: 0)
        result = 31 * result + colorPallete.hashCode()
        return result
    }
}
